src/Lib/Configuration.php

```diff
@@ -208,8 +221,19 @@ public function getHealthCheckIntervalSeconds()
     /**
      * @return LoggerInterface
      */
-    public function getLogger()
+    public function getLogger(): LoggerInterface
     {
         return $this->logger;
     }
+
+    /**
+     * @return HttpClient
+     */
+    public function getClient(): HttpClient
+    {
+        return new HttpMethodsClient(
+            $this->client ?? HttpClientDiscovery::find(),
+            MessageFactoryDiscovery::find()
```

Comment:
The second parameter are from Http\Message\RequestFactory interface and with `Psr17FactoryDiscovery::findRequestFactory()` it's Psr\Http\Message\RequestFactoryInterface. It's not the same interface.

I've got this error : 

> Fatal error: Uncaught TypeError: Argument 2 passed to Http\Client\Common\HttpMethodsClient::__construct() must implement interface Http\Message\RequestFactory

Maybe it will be fixed in a next release.